nodejs/eks/cluster.ts

```diff
@@ -313,8 +313,8 @@ export function getRoleProvider(
     parent?: pulumi.ComponentResource,
     provider?: pulumi.ProviderResource,
 ): CreationRoleProvider {
-    const partition = pulumi.output(aws.getPartition({ parent })).partition;
-    const accountId = pulumi.output(aws.getCallerIdentity({ parent })).accountId;
+    const partition = pulumi.output(aws.getPartition({})).partition;
+    const accountId = pulumi.output(aws.getCallerIdentity({})).accountId;
```

Comment:
Why are we no longer passing parent? Not passing parent causes `getRoleProvider` to fail when default providers are disabled.